front/src/common/BootstrapSNCF/SelectImprovedSNCF.tsx

```diff
@@ -1,55 +1,83 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, ReactNode, useCallback } from 'react';
+import nextId from 'react-id-generator';
+import { isObject } from 'lodash';
+
+import { SelectOptionObject } from './SelectSNCF';
 import './SelectImprovedSNCF.scss';
 
-import PropTypes from 'prop-types';
-import nextId from 'react-id-generator';
+interface SelectProps<T> {
+  inline?: boolean;
+  label?: ReactNode;
+  options: T[];
+  value?: T;
+  onChange: (value?: T) => void;
+  sm?: boolean;
+  withSearch?: boolean;
+  withNewValueInput?: boolean;
+  addButtonTitle?: string;
+  bgWhite?: boolean;
+  dataTestId?: string;
+}
 
-export default function SelectImprovedSNCF(props) {
-  const {
-    inline,
-    title,
-    options,
-    selectedValue,
-    onChange,
-    sm,
-    withSearch,
-    withNewValueInput,
-    addButtonTitle,
-    bgWhite,
-    dataTestId,
-  } = props;
+function SelectImproved<T extends string | SelectOptionObject>({
+  inline = false,
+  label,
+  options,
+  value,
+  onChange,
+  sm = false,
+  withSearch = false,
+  withNewValueInput = false,
+  addButtonTitle = 'Add',
+  bgWhite = false,
+  dataTestId = '',
```

Comment:
No need to set default value to false
```suggestion
  inline,
  label,
  options,
  value,
  onChange,
  sm,
  withSearch,
  withNewValueInput,
  addButtonTitle = 'Add',
  bgWhite,
  dataTestId = '',
```